test/support/src/vfs_helpers.cc

```diff
@@ -415,5 +438,137 @@ std::string TemporaryDirectoryFixture::create_temporary_array(
   return array_uri;
 }
 
+VFSTestBase::VFSTestBase(
+    const std::vector<size_t>& test_tree, const std::string& prefix)
+    : test_tree_(test_tree)
+    , compute_(4)
+    , io_(4)
+    , vfs_(
+          &tiledb::test::g_helper_stats,
+          &io_,
+          &compute_,
+          tiledb::test::create_test_config())
+    , prefix_(prefix)
+    , temp_dir_(tiledb::test::test_dir(prefix_))
+    , is_supported_(vfs_.supports_uri_scheme(temp_dir_)) {
+  // TODO: Throw when we can provide a list of supported filesystems to Catch2.
+}
+
+VFSTestBase::~VFSTestBase() {
+  if (vfs_.supports_uri_scheme(temp_dir_)) {
+    bool is_dir = false;
+    vfs_.is_dir(temp_dir_, &is_dir).ok();
+    if (is_dir) {
+      vfs_.remove_dir(temp_dir_).ok();
+    }
+  }
+}
+
+VFSTest::VFSTest(
+    const std::vector<size_t>& test_tree, const std::string& prefix)
+    : VFSTestBase(test_tree, prefix) {
+  setup_test();
+}
+
+void VFSTest::test_ls_recursive(
```

Comment:
This function should not be defined in this file at all, which likely means it doesn't belong in this class. 

Tests for the filtered lists belong in the translation units that use them. In this case it's `unit-vfs.cc`. In order to get the definition there, it means that the present `class VFSTest` should be the base class of a class derived from it and defined in `unit-vfs.cc`.